### ISSUE TITLE
HCS: Consider cached on-the-fly scale pyramids

### ISSUE THREAD

**Issue by tischi** [mobie/mobie-viewer-fiji#991]
It could be interesting to compute a downscaled version of single resolution images while loading and then cache them, while potentially throwing away the high-resolution data. 

Probably it could be good to do this already at the loading level,  for example when building the `Image` from `SpimData`.

To make this efficient may also need to take care that the high-resolution source data is GC'ed to free up memory.

**Comment by ThomasMicroscopy** [comment]
Yes, maybe not a bad idea to have overview thumbnails created on the fly, but maybe even keep them for further use later?
